cmd/attest.go

```diff
@@ -0,0 +1,173 @@
+--output flag provides file name to store attestation report in the format specified
+by --format flag.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		rwc, err := openTpm()
+		if err != nil {
+			return err
+		}
+		defer rwc.Close()
+
+		var attestationKey *client.Key
+		algoToCreateAK, ok := attestationKeys[key]
+		if !ok {
+			return fmt.Errorf("key should be either AK or gceAK")
+		}
+		createFunc := algoToCreateAK[keyAlgo]
+		attestationKey, err = createFunc(rwc)
+		if err != nil {
+			return fmt.Errorf("failed to create attestation key: %v", err)
+		}
+		defer attestationKey.Close()
+
+		attestOpts := client.AttestOpts{}
+		attestOpts.Nonce = nonce
+		if len(teeNonce) != 0 {
+			attestOpts.TEENonce = teeNonce
+			attestOpts.TEEDevice, err = client.CreateSevSnpDevice()
+			if err != nil {
+				return fmt.Errorf("failed to collect TEE attestation report: %v", err)
+			}
+		}
+
+		attestOpts.TCGEventLog, err = client.GetEventLog(rwc)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve TCG Event Log: %w", err)
+		}
+
+		attestation, err := attestationKey.Attest(attestOpts)
+		if err != nil {
+			return fmt.Errorf("failed to collect attestation report : %v", err)
+		}
+
+		if key == "gceAK" {
+			instanceInfo, err := getInstanceInfoFromMetadata()
+			if err != nil {
+				return err
+			}
+			attestation.InstanceInfo = instanceInfo
+		}
+
+		var out []byte
+		if format == "binarypb" {
+			out, err = proto.Marshal(attestation)
+			if err != nil {
+				return fmt.Errorf("failed to marshal attestation proto: %v", attestation)
+			}
+		} else if format == "textproto" {
+			out = []byte(marshalOptions.Format(attestation))
+		} else {
+			return fmt.Errorf("format should be either binarypb or textproto")
+		}
+		if _, err := dataOutput().Write(out); err != nil {
+			return fmt.Errorf("failed to write attestation report: %v", err)
+		}
+		return nil
+	},
+}
+
+func getInstanceInfoFromMetadata() (*attest.GCEInstanceInfo, error) {
+
+	var err error
+	instanceInfo := &attest.GCEInstanceInfo{}
+
+	instanceInfo.ProjectId, err = metadata.ProjectID()
+	if err != nil {
+		return nil, err
+	}
+
+	projectNumber, err := metadata.NumericProjectID()
+	if err != nil {
+		return nil, err
+	}
+	instanceInfo.ProjectNumber, err = strconv.ParseUint(projectNumber, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceInfo.Zone, err = metadata.Zone()
+	if err != nil {
+		return nil, err
+	}
+
+	instanceID, err := metadata.InstanceID()
+	if err != nil {
+		return nil, err
+	}
+	instanceInfo.InstanceId, err = strconv.ParseUint(instanceID, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceInfo.InstanceName, err = metadata.InstanceName()
+	if err != nil {
+		return nil, err
+	}
+
+	return instanceInfo, err
+}
+
+func addKeyFlag(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(&key, "key", "AK", "indicates type of attestation key to use <gceAK|AK>")
+}
+func addTeeNonceflag(cmd *cobra.Command) {
```

Comment:
I notice this isn't added in verify.go. How does that test pass?

Reply:
Thanks. Please move this to flags.go

Reply:
Sure